grpc/src/client/load_balancing/round_robin.rs

```diff
@@ -0,0 +1,1319 @@
+    // picker.
+    async fn verify_roundrobin_ready_picker(
+        rx_events: &mut mpsc::UnboundedReceiver<TestEvent>,
+    ) -> Arc<dyn Picker> {
+        println!("verify ready picker");
+        match rx_events.recv().await.unwrap() {
+            TestEvent::UpdatePicker(update) => {
+                println!(
+                    "connectivity state for ready picker is {}",
+                    update.connectivity_state
+                );
+                assert!(update.connectivity_state == ConnectivityState::Ready);
+                let req = test_utils::new_request();
+                match update.picker.pick(&req) {
+                    PickResult::Pick(pick) => update.picker.clone(),
+                    other => panic!("unexpected pick result {}", other),
+                }
+            }
+            other => panic!("unexpected event {:?}", other),
+        }
+    }
+
+    // Verifies that the channel moves to TRANSIENT_FAILURE state with a picker
+    // that returns an error with the given message. The error code should be
+    // UNAVAILABLE..
+    //
+    // Returns the picker for tests to make more picks, if required.
+    async fn verify_transient_failure_picker(
+        rx_events: &mut mpsc::UnboundedReceiver<TestEvent>,
+        want_error: String,
+    ) -> Arc<dyn Picker> {
+        let picker = match rx_events.recv().await.unwrap() {
+            TestEvent::UpdatePicker(update) => {
+                assert!(update.connectivity_state == ConnectivityState::TransientFailure);
+                let req = test_utils::new_request();
+                match update.picker.pick(&req) {
+                    PickResult::Fail(status) => {
+                        assert!(status.code() == tonic::Code::Unavailable);
+                        dbg!(status.message());
+                        dbg!(&want_error);
+                        assert!(status.message().contains(&want_error));
+                        update.picker.clone()
+                    }
+                    other => panic!("unexpected pick result {}", other),
+                }
+            }
+            other => panic!("unexpected event {:?}", other),
+        };
+        picker
+    }
+
+    // Verifies that the LB policy requests re-resolution.
+    async fn verify_resolution_request(rx_events: &mut mpsc::UnboundedReceiver<TestEvent>) {
+        println!("verifying resolution request");
+        match rx_events.recv().await.unwrap() {
+            TestEvent::RequestResolution => {}
+            other => panic!("unexpected event {:?}", other),
+        };
+    }
+
+    async fn verify_no_activity(rx_events: &mut mpsc::UnboundedReceiver<TestEvent>) {
+        tokio::select! {
+            _ = tokio::time::sleep(DEFAULT_TEST_SHORT_TIMEOUT) => {}
+            event = rx_events.recv() => {
+                panic!("unexpected event {:?}", event.unwrap());
+            }
+        }
+    }
+
+    #[test]
+    fn roundrobin_builder_name() -> Result<(), String> {
```

Comment:
This test seems to be testing that the name string is "round_robin". I'm not sure if it's very useful.